In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

In [2]:
df=pd.read_csv('large_time_waste_data.csv')

In [3]:
df.head()

age  occupation  hours_worked  hours_leisure  screen_time  commute_time  \
0   56           4             7              6            3             1   
1   46           1             5              7            3             2   
2   32           0            11              5            6             2   
3   60           4            11              3            5             1   
4   25           0            11              2            3             1   

   time_wasted  
0            4  
1            0  
2            0  
3            4  
4            0

In [4]:
df.tail()

age  occupation  hours_worked  hours_leisure  screen_time  commute_time  \
9995   33           1             5              5            5             0   
9996   29           0             2              2            1             0   
9997   18           0            11              3            6             2   
9998   25           0             1              1            0             0   
9999   30           4             4              5            2             2   

      time_wasted  
9995            3  
9996            0  
9997            2  
9998            4  
9999            1

In [5]:
df.info

<bound method DataFrame.info of       age  occupation  hours_worked  hours_leisure  screen_time  commute_time  \
0      56           4             7              6            3             1   
1      46           1             5              7            3             2   
2      32           0            11              5            6             2   
3      60           4            11              3            5             1   
4      25           0            11              2            3             1   
...   ...         ...           ...            ...          ...           ...   
9995   33           1             5              5            5             0   
9996   29           0             2              2            1             0   
9997   18           0            11              3            6             2   
9998   25           0             1              1            0             0   
9999   30           4             4              5            2             2

In [6]:
df.describe()

age    occupation  hours_worked  hours_leisure   screen_time  \
count  10000.000000  10000.000000  10000.000000   10000.000000  10000.000000   
mean      41.054000      1.996000      5.474800       3.514100      3.521100   
std       13.484104      1.421262      3.448703       2.272029      2.294228   
min       18.000000      0.000000      0.000000       0.000000      0.000000   
25%       30.000000      1.000000      3.000000       2.000000      2.000000   
50%       41.000000      2.000000      5.000000       4.000000      4.000000   
75%       53.000000      3.000000      8.000000       5.250000      6.000000   
max       64.000000      4.000000     11.000000       7.000000      7.000000   

       commute_time  time_wasted  
count  10000.000000  10000.00000  
mean       1.005700      2.00510  
std        0.811377      1.42558  
min        0.000000      0.00000  
25%        0.000000      1.00000  
50%        1.000000      2.00000  
75%        2.000000      3.00000  
max        2.000000      4.00000

In [8]:
df.shape

(10000, 7)

In [9]:
df.columns

Index(['age', 'occupation', 'hours_worked', 'hours_leisure', 'screen_time',
       'commute_time', 'time_wasted'],
      dtype='object')

In [10]:
df.isnull()

age  occupation  hours_worked  hours_leisure  screen_time  \
0     False       False         False          False        False   
1     False       False         False          False        False   
2     False       False         False          False        False   
3     False       False         False          False        False   
4     False       False         False          False        False   
...     ...         ...           ...            ...          ...   
9995  False       False         False          False        False   
9996  False       False         False          False        False   
9997  False       False         False          False        False   
9998  False       False         False          False        False   
9999  False       False         False          False        False   

      commute_time  time_wasted  
0            False        False  
1            False        False  
2            False        False  
3            False        False  
4            False        False  
...            ...          ...  
9995         False        False  
9996         False        False  
9997         False        False  
9998         False        False  
9999         False        False  

[10000 rows x 7 columns]

In [11]:
df.notnull()

age  occupation  hours_worked  hours_leisure  screen_time  \
0     True        True          True           True         True   
1     True        True          True           True         True   
2     True        True          True           True         True   
3     True        True          True           True         True   
4     True        True          True           True         True   
...    ...         ...           ...            ...          ...   
9995  True        True          True           True         True   
9996  True        True          True           True         True   
9997  True        True          True           True         True   
9998  True        True          True           True         True   
9999  True        True          True           True         True   

      commute_time  time_wasted  
0             True         True  
1             True         True  
2             True         True  
3             True         True  
4             True         True  
...            ...          ...  
9995          True         True  
9996          True         True  
9997          True         True  
9998          True         True  
9999          True         True  

[10000 rows x 7 columns]

In [24]:
from sklearn.preprocessing import StandardScaler
std_scaler = StandardScaler()

In [28]:
y1=df['time_wasted']
x1=df.drop('time_wasted',axis=1)

In [29]:
x1_scaled = std_scaler.fit_transform(x1)
x1_scaled

array([[ 1.10847166,  1.41008487,  0.4422753 ,  1.0941867 , -0.22714661,
        -0.00702544],
       [ 0.36682061, -0.70082063, -0.13768182,  1.53434387, -0.22714661,
         1.22550822],
       [-0.67149086, -1.40445579,  1.60218953,  0.65402953,  1.08054831,
         1.22550822],
       ...,
       [-1.70980233, -1.40445579,  1.60218953, -0.2262848 ,  1.08054831,
         1.22550822],
       [-1.1906466 , -1.40445579, -1.29759605, -1.10659913, -1.53484152,
        -1.23955911],
       [-0.81982107,  1.41008487, -0.42766038,  0.65402953, -0.66304491,
         1.22550822]])

In [30]:
from sklearn.model_selection import train_test_split
x1_train, x1_test, y1_train, y1_test = train_test_split(x1_scaled,y1,test_size=0.2,random_state= 42,stratify =y1)

In [31]:
x1_train.shape , x1_test.shape

((8000, 6), (2000, 6))

In [32]:
from sklearn.linear_model import LogisticRegression
from sklearn.tree import DecisionTreeClassifier
from sklearn.ensemble import RandomForestClassifier
from sklearn.ensemble import ExtraTreesClassifier
from sklearn.svm import SVC
from sklearn.neighbors import KNeighborsClassifier
from sklearn.ensemble import GradientBoostingClassifier
from sklearn.naive_bayes import GaussianNB
from sklearn.ensemble import AdaBoostClassifier

In [33]:
LOR = LogisticRegression()
DT = DecisionTreeClassifier()
RF = RandomForestClassifier()
ETC = ExtraTreesClassifier()
SVM = SVC()
KNN = KNeighborsClassifier()
GBC = GradientBoostingClassifier()
ABC = AdaBoostClassifier()
NB = GaussianNB()

In [34]:
from sklearn.model_selection import cross_val_score

In [35]:
cls_models = [LOR,DT,RF,ETC,SVM,KNN,GBC,ABC,NB]
features = x1_scaled
labels = y1
CV = 5
accu_list = [] #Accuracy List
ModelName = [] #Model Name List

for model in cls_models:
  model_name = model.__class__.__name__
  accuracies = cross_val_score(model,features,labels,scoring = 'accuracy',cv = CV)
  accu_list.append(accuracies.mean()*100)
  ModelName.append(model_name)

model_acc_df = pd.DataFrame({"Model" : ModelName , "Cross_Val_Accuracy" : accu_list})
model_acc_df

Model  Cross_Val_Accuracy
0          LogisticRegression               20.61
1      DecisionTreeClassifier               19.81
2      RandomForestClassifier               19.41
3        ExtraTreesClassifier               19.32
4                         SVC               19.97
5        KNeighborsClassifier               20.07
6  GradientBoostingClassifier               19.67
7          AdaBoostClassifier               20.36
8                  GaussianNB               20.54

In [39]:
model = LogisticRegression()
model.fit(x1_train , y1_train)

LogisticRegression()

In [40]:
x1=df.drop(columns=['time_wasted'])
y1=df['time_wasted']
from sklearn.model_selection import cross_val_score, train_test_split
def classify(model, x1, y1):
   x1_train, x1_test, y1_train, y1_test=train_test_split(x1, y1, test_size=0.25, random_state=42)
#train the model
   model.fit(x1_train, y1_train)
   print("Accuracy:", model.score(x1_test, y1_test)*100)
#cross-validation
   score=cross_val_score(model, x1, y1, cv=5) 
   print("CV Score:", np.mean(score) *100)

In [41]:
from sklearn.linear_model import LogisticRegression
model= LogisticRegression()
classify(model, x1, y1)

/home/rgukt/anaconda3/lib/python3.11/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning:

lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression



Accuracy: 20.4


/home/rgukt/anaconda3/lib/python3.11/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning:

lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression

/home/rgukt/anaconda3/lib/python3.11/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning:

lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression

/home/rgukt/anaconda3/lib/

CV Score: 20.57


/home/rgukt/anaconda3/lib/python3.11/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning:

lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression



<h3>The Logistic Regression model is predicting the time wasted with an accuracy of 20.4%</h3>